### PR TITLE
controllers: add unit-tests for versioned migration

### PR DESCRIPTION
**Pull request by datdao**

### CODE REVIEW

File: controllers/atlasmigration_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+
+// newAtlasMigrationReconcilerMock returns a AtlasMigrationReconciler with a fake client
+// please create a new namespace for your suite tests
+func mockMigrationReconciler(ns string, objs ...client.Object) *AtlasMigrationReconciler {
+	scheme := runtime.NewScheme()
+	v1alpha1.AddToScheme(scheme)
+	corev1.AddToScheme(scheme)
+
+	switch ns {
+	case "default":
```
Comment (giautm) [inline controllers/atlasmigration_controller_test.go:48]: why only check `default` namespace?
```suggestion
	switch ns {
	case "default":
```

File: controllers/atlasmigration_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+	file, cleanup, err := migrate.render()
+	require.NoError(t, err)
+	parse, err := url.Parse(file)
+	require.NoError(t, err)
+	fileContent, err := os.ReadFile(parse.Path)
+	require.NoError(t, err)
+	require.FileExists(t, parse.Path)
+	require.EqualValues(t, `
+
+env {
```
Comment (giautm) [inline controllers/atlasmigration_controller_test.go:358]: remove empty lines in the template

File: controllers/atlasmigration_controller_test.go (hunk 1 above)
Comment (datdao) [reply controllers/atlasmigration_controller_test.go]: removed, but the idea behide is producing local env by args

**Review by giautm — APPROVED** [review]

File: controllers/atlasmigration_controller_test.go [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+		},
+	})
+	require.NoError(t, err)
+	require.Equal(t, tt.dburl, amd.URL)
+	require.Equal(t, "https://atlasgo.io/", amd.Cloud.URL)
+	require.Equal(t, "my-project", amd.Cloud.Project)
+	require.Equal(t, "my-token", amd.Cloud.Token)
+	require.Equal(t, "my-remote-dir", amd.Cloud.RemoteDir.Name)
+	require.Equal(t, "my-remote-tag", amd.Cloud.RemoteDir.Tag)
+	cleanUp()
```
Comment (giautm) [inline controllers/atlasmigration_controller_test.go:192]: use `defer` ;)

**Pull request merged**